iphone/Maps/Classes/CustomViews/MapViewControls/TrackRecordingButtonViewController.swift

```diff
@@ -38,31 +39,29 @@ final class TrackRecordingViewController: MWMViewController {
     fatalError("init(coder:) has not been implemented")
   }
 
-  override func viewDidAppear(_ animated: Bool) {
-    super.viewDidAppear(animated)
-    UIView.transition(with: self.view,
-                      duration: kDefaultAnimationDuration,
-                      options: .transitionCrossDissolve,
-                      animations: {
-      self.button.isHidden = false
-    })
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    // async is for smoother appearance
+    DispatchQueue.main.asyncAfter(deadline: .now() + kDefaultAnimationDuration) {
+      self.setState(self.state, completion: nil)
+    }
```

Comment:
nit: Docs say:
> Specifying the current time is less efficient than calling the [async(execute:)](https://developer.apple.com/documentation/dispatch/dispatchqueue/async(execute:)) method directly.

Will this work?

```suggestion
    DispatchQueue.main.async {
      self.setState(self.state, completion: nil)
    }
```

Reply:
removed!